daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -415,24 +439,32 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
   }
 
   /**
-   * Creates a new instance of a variable
+   * Creates a new instance of a variable with default value
    */
-  def newVariableInstance(vrd: VariableRuntimeData, state: ParseOrUnparseState) = {
+  def newVariableInstance(vrd: VariableRuntimeData, defaultValue: DataValuePrimitive) = {
     val varQName = vrd.globalQName
     val abuf = vTable.get(varQName)
     Assert.invariant(abuf.isDefined)
 
-    if (vrd.maybeDefaultValueExpr.isDefined) {
-      val defaultValue = DataValue.unsafeFromAnyRef(vrd.maybeDefaultValueExpr.get.evaluate(state))
-      abuf.get += vrd.createVariableInstance(VariableUtils.convert(defaultValue.getAnyRef.toString, vrd, vrd))
-    } else
-      abuf.get += vrd.createVariableInstance()
+    abuf.get += vrd.createVariableInstance(VariableUtils.convert(defaultValue.getAnyRef.toString, vrd, vrd))
   }
```

Comment:
I wonder if this could be made more clear if the defaultValue wasn't passed into this, and add a new function that's something like setDefaultValue()? That way it's very clear when the NVI is actually created, and also how the default value is set, and what state changes when it's set? Not sure off hand, but just a suggestion.